OpenRA.Mods.Common/Widgets/WidgetUtils.cs

```diff
@@ -270,6 +271,24 @@ public static void TruncateButtonToTooltip(ButtonWidget button, string text)
 			else
 				button.GetTooltipText = null;
 		}
+
+		public static void AddSuffixToPlayerNameLabel(LabelWidget label, Player p)
+		{
+			var client = p.World.LobbyInfo.ClientWithIndex(p.ClientIndex);
+			var playerNameFont = Game.Renderer.Fonts[label.Font];
+			var suffixLength = new CachedTransform<string, int>(s => playerNameFont.Measure(s).X);
+			var name = new CachedTransform<Pair<string, string>, string>(c =>
+				WidgetUtils.TruncateText(c.First, label.Bounds.Width - suffixLength.Update(c.Second), playerNameFont));
+
+			label.GetText = () =>
+			{
+				var suffix = p.WinState == WinState.Undefined ? "" : " (" + p.WinState + ")";
+				if (client != null && client.State == Session.ClientState.Disconnected)
+					suffix = " (Gone)";
+
+				return name.Update(Pair.New(p.PlayerName, suffix)) + suffix;
```

Comment:
This looks odd to me. Isn't the whole point of having a `CachedTransform` that we do not create a new string on every `GetText`? The `+ suffix` would probably better remain as `+ c.Second` in the transform.

Reply:
I copied the code from here: https://github.com/OpenRA/OpenRA/blob/bleed/OpenRA.Mods.Common/Widgets/Logic/Ingame/GameInfoStatsLogic.cs#L108. A slightly different version also exists here: https://github.com/OpenRA/OpenRA/blob/bleed/OpenRA.Mods.Common/Widgets/Logic/Ingame/ObserverStatsLogic.cs#L542 (this I replaced with the new util function). And it still exists here: https://github.com/OpenRA/OpenRA/blob/41657dd29162459bcaf7e48dd549ca7974a3d2b5/OpenRA.Mods.Common/Widgets/Logic/Ingame/GameInfoStatsLogic.cs#L155 (could not replace it because the util needs a reference to the player). 

How exactly should this be fixed? I one of the existing approaches correct?

Reply:
The first one is correct.